sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/AnnouncementFragment.java

```diff
@@ -61,4 +97,26 @@ public void initializeViews() {
     public void setAnnouncement(Announcement announcement) {
         this.announcement = announcement;
     }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        EventBus.getDefault().register(this);
+    }
+
+//    @Override
+//    public void onResume() {
+//        Requests.Announcements.with(getActivity()).makeShowRequest(announcement);
+//        super.onResume();
+//    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        EventBus.getDefault().unregister(this);
+    }
+
+    public void onEvent(AnnouncementEvent event) {
+
```

Comment:
here you should call initializeview again, incase someone edited the announcement in between activities. set the fragment's announcement to the updated one. in fact, you should probably just have a initializeAnnouncement() method (separate it from initializeView()), and do all the setting of text and stuf fthere.